### PR TITLE
Adding job feed + perfomance disscussion

### PR DESCRIPTION
**Pull request by houssem-yahiaoui**
This PR have two part : 

**## The first one :** 

Possible tested solution for issue #9 proposed by Riad, consist of returning max of top 5 jobs from dyalna.com web site,

The command will return top 5 objects within the scope of those 2 options: 

1. If we ask the bot about jobs in a channel different then #jobs, the bot will ask the user to ask her over #jobs channel.
2. Via direct message.

Last i want to mention that the 'fallback' option over the showcasing  options is required by slack, the why is that sometimes, if we do ask the bot for some stuff then not being **physically** present over slack a **notification** will be returned as part of the UX that slack use, so the fallback will give us a nicely looking fallback message of what's happening over **slack** now.

benefits of fallback over slack attachments

**## Second one :** 

some loops enhancements regarding some perfomance over the data.body.forEach, i've replaced it with a normal loop taking on consideration this [jsperf benchmark.](https://jsperf.com/for-loop-test-more)

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -1,3 +1,3 @@
-# Generated at https://my.slack.com/services/new/bot
-# API_TOKEN =
-# API_TOKEN_DEV =
+-# Generated at https://my.slack.com/services/new/bot
```
Comment (IOAyman) [inline .env.sample:1]: What are these for? 😕

File: package.json [hunk 2]
```diff
@@ -30,7 +30,9 @@
     "botkit": "^0.4.9",
     "bunyan": "^1.8.5",
     "dotenv": "^4.0.0",
+    "install": "^0.8.4",
```
Comment (IOAyman) [inline package.json]: What're you doing with that package?

File: package.json [hunk 3]
```diff
@@ -30,7 +30,9 @@
...
     "moment": "^2.17.1",
+    "npm": "^4.1.1",
```
Comment (IOAyman) [inline package.json]: How about this one?

**Review by IOAyman — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (houssem-yahiaoui) [reply package.json]: nothing really probably from that package removal i made.

**Comment by IOAyman** [comment]
You might want to regenerate the `yarn.lock` file as well @houssem-yahiaoui ;)

**Comment by houssem-yahiaoui** [comment]
Yes, i'm on it :)

**Comment by IOAyman** [comment]
Concerning that performance test, I'm curious if there's a write-up digging more about those tests?
Another question is, are those tests for browsers only? Or does it include node as well 😕

**Comment by houssem-yahiaoui** [comment]
@IOAyman I guess the test was made in a web page to begin with and jsperf act on benchmarking regarding a a web page, but i do believe even if two ecosystem are different (web/node) we  the test was made over chrome using the v8 engine so it is there present, probably same thing applied to node v8 ! but that just me.

**Review by IOAyman — APPROVED** [review]
LGTM 👍

Closes #9

**Comment by IOAyman** [comment]
Let's take the conversation over the the [Slack](algeriatech.slack.com)

File: src/controllers/dyalna.js [hunk 4]
```diff
@@ -1,31 +1,62 @@
...
 
-
 const news = (bot, msg) => {
   superagent
     .get('https://dyalna.com/api/post/top')
     .then(data => {
       bot.startConversation(msg, (err, convo) => {
         convo.say('Sraw bezzaf swala7, voici les top news :')
-        data.body.forEach(post => {
+        for (let i = 0, len = data.body.length; i <= len; i++) {
```
Comment (youknowriad) [inline src/controllers/dyalna.js:9]: For each is better in my opinion

File: src/controllers/dyalna.js [hunk 5]
```diff
@@ -1,31 +1,62 @@
...
+      })
+    })
+}
+
+const jobs = (bot, msg) => {
+  superagent
+    .get('http://jobs.dyalna.com/jobs/api.php/job?category=1&page=1')
+    .then(data => {
+      bot.startConversation(msg, (err, convo) => {
+        if (msg.channel == 'C3NRY5JP2' || msg.event == 'direct_message') {
```
Comment (youknowriad) [inline src/controllers/dyalna.js:34]: I think there's a constant for direct_message

File: src/controllers/dyalna.js [hunk 6]
```diff
@@ -1,31 +1,62 @@
...
+          convo.say('Undi khir we khmir khayer li tnaseb tes compétences ! :')
+          for (let i = 0, len = 5; i < len; i++) {
+            convo.say({
+              'attachments': [{
+                'fallback': 'Dyalna Jobs',
+                'color': '#36a64f',
+                'author_name': 'Dyalna Jobs',
+                'author_link': 'dyalna.com',
+                'title': `${ data.body[i].title }`,
+                'title_link': `${ data.body[i].url }`,
```
Comment (youknowriad) [inline src/controllers/dyalna.js:44]: I would rather use the Dyalna Jobs link (to thank Dyalna Jobs for the centralized API :), I know it's my website )

**Comment by Ingeskhaled** [comment]
Bonjour les jeunes !! par ce message je m'adresse à Yahioui Houssem. Je te prise de répondre au téléphone d'ici ce soir sinon je commencerais par déposer plainte pour vol qualifié.

Je sais très bien que tu vois mes appels  !!

**Comment by ghost** [comment]
Bonjour les gars, j'ai publié un message par erreur  tout à l'heure
merci de ne pas prendre en compte

**Pull request merged**